src/rust/engine/process_execution/src/local.rs

```diff
@@ -266,9 +268,15 @@ impl super::CommandRunner for CommandRunner {
     let argv = req.argv;
     let req_description = req.description;
     let maybe_jdk_home = req.jdk_home;
+    let local_only_scratch_files = req.local_only_scratch_files;
+    let workunit_store2 = workunit_store.clone();
+
     self
       .store
       .materialize_directory(workdir_path.clone(), req.input_files, workunit_store)
+      .and_then(move |_metadata| {
```

Comment:
Please add a test which depends on having the digest materialized, which shows that it is materialized

Reply:
Added in https://github.com/pantsbuild/pants/pull/8282/commits/b1f32f8bc05c4d726b27be1c6c57c52ae58d4a71